nixos/modules/virtualisation/xen-dom0.nix

```
@@ -610,10 +709,12 @@ in
         message = "Xen is currently not supported on ${pkgs.stdenv.hostPlatform.system}.";
       }
       {
-        assertion =
-          config.boot.loader.systemd-boot.enable
-          || (config.boot ? lanzaboote) && config.boot.lanzaboote.enable;
-        message = "Xen only supports booting on systemd-boot or Lanzaboote.";
+        assertion = cfg.efiBoot.enable || cfg.biosBoot.enable;
+        message = "Either efiBoot.enable or biosBoot.enable must be enabled";
```

```suggestion
        message = "Only one of 'virtualisation.xen.boot.efi.enable' or 'virtualisation.xen.boot.bios.enable' must be enabled.";
```
Shouldn't this be "***only one of 'virtualisation.xen.boot.efi.enable' or 'virtualisation.xen.boot.bios.enable' must be enabled*"? The assertion will allow for someone enabling *both* BIOS boot and EFI boot.

---

That was initially an intended configuration option yes, to cater to bootloaders capable for booting both, for users who might be migrating from bios to EFI booting or CSM to pure EFI booting. I didn't end up taking the route where this was managed by this module, and install relegated this to the bootloader module by using a pure bootspec extension approach in this PR. So this has been deleted, I've pushed a commit.